interop/xds/server/server.go

```diff
@@ -78,8 +93,89 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}
+
+behavior:
+	for _, headerVal := range getRPCBehaviorMetadata(ctx) {
+		switch {
+		//If the value matches "sleep-<int>", the server should wait the specified number of seconds before resuming behavior matching and RPC processing.
+		case strings.HasPrefix(headerVal, sleepPfx):
+			{
+				sleep, err := strconv.Atoi(headerVal[len(sleepPfx):])
+				if err != nil {
+					return nil, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header %v, must be '%v<int>' instead", headerVal, sleepPfx)
+				}
+
+				time.Sleep(time.Duration(sleep) * time.Second)
+			}
+		// If the value matches "keep-open", the server should never respond to the request and behavior matching ends.
+		case strings.HasPrefix(headerVal, keepOpenVal):
+			{
+				<-ctx.Done()
```

Comment:
There should probably be a `break forLoop` after this receive returns?

Reply:
shall I add a `return nil, nil` instead?
assuming it is fed the request's context, if the Done channel returns it would mean that the request has been closed by the client, so the return values will be of no use anyways?

Reply:
Returning nils would still be the way to go, because the spec says: `the server should never respond to the request and behavior matching ends`. So, we don't want to go back to the loop once the context expires.